### DISCUSSION TITLE
Reason for incomplete tags for maxspeed quest?

### DISCUSSION THREAD

**Discussion by NathanARF** [streetcomplete/StreetComplete#5131] [Q&A]
I've noticed when marking a single carriageway road segment in the UK as the National Speed Limit, that only the tag `maxspeed:type=GB:nsl_single` is applied and not `maxspeed=60 mph` as well. Why is this?

**Comment by mnalis** [comment]
See https://github.com/streetcomplete/StreetComplete/issues/3812#issuecomment-1060055618.

Short answer: because StreetComplete **only tags what the user _sees_**. It requires no knowledge of local legislation from users (after all, non-drivers, children, and people just visiting UK might be using StreetComplete in the UK). 

There are tools like https://westnordost.github.io/osm-legal-default-speeds/ to infer the actual speed limit from what is tagged (and it is much more complex then just `maxspeed=60mph`, see [example](https://westnordost.github.io/osm-legal-default-speeds/#tags=maxspeed%3Atype%3DGB%3Ansl_single&cc=GB))

Also, think what happens when law changes (ref: https://github.com/streetcomplete/StreetComplete/issues/4282#issuecomment-1607879594)

See also:
- https://github.com/streetcomplete/StreetComplete/issues/1179
- https://github.com/streetcomplete/StreetComplete/issues/1878

**Comment by NathanARF** [comment]
But the `maxspeed:type=*` tag implies a speed limit, either 30, 60 or 70 mph, and this accompanying tag is specified in the wiki page: "`maxspeed:type=*` describes how a speed limit has been set into force. It should be accompanied by a `maxspeed=*` tag." Is the wiki incorrect with respect to this? Additionally, if the national speed limit were to change, then would it not be easy to change tags across the country, as has been done nationally with the change of name of the road operator 'Highways England' to 'National Highways'?

**Reply by mnalis** [reply]
> "[maxspeed:type=*](https://wiki.openstreetmap.org/wiki/Key:maxspeed:type) describes how a speed limit has been set into force. It should be accompanied by a maxspeed=* tag."

Well, I just link you to previous explanations, have you read them too? I do not really hold hard opinion on it (I don't drive car, so I find a maxspeed tagging useless either way for my cycling and walking :smile:). 
So it would seem to me that wiki _"should be accompanied"_ is at least disputed (I've added a note to its talk page now, linking here).
As a person coming from database world, I however do see how having such duplicated (and potentially conflicting) sources of implicit/explicit values (and thus breaking [2NF](https://en.wikipedia.org/wiki/Second_normal_form)) sounds like a bad idea.

> Additionally, if the national speed limit were to change

As far as I understand, problem with that part is (if you read those discussions) that since SC doesn't ask mapper to manually input speed value (as it doesn't expect them to know it, and only asks them to tag what they **_see_**), only alternative would be if SC had a world-wide curated database of implied speed limits (like [osm-legal-default-speeds](https://westnordost.github.io/osm-legal-default-speeds/)  does). The problem with that is, even when law changes and someone does automated edit to update existing values, any existing non-updated version of StreetComplete would continue to put **wrong values on the map for a long time** (until they are upgraded - if ever), which I can imagine would be quite annoying and problematic (and thus make StreetComplete unpopular).

> then would it not be easy to change tags across the country

Well, if it is easy and safe to do with automated edits, then couldn't they just be setup to automatically add not only `maxspeed=60mph`, but also:

```
maxspeed=60 mph
maxspeed:bus=50 mph
maxspeed:conditional=50 mph @ (trailer); 20 mph @ (trailers>=2)
maxspeed:hgv=50 mph
maxspeed:motorhome:conditional=50 mph @ (emptyweight>3.05)
```

on all roads tagged with `maxspeed:type=GB:nsl_single`. Because, bot periodically running such automated edit would have same result, and less potential problems, then if StreetComplete were to do it (as noted above).

Because, there is another issue if you add **only** `maxspeed=60mph`, you're making map **worse** -- it (without other maxspeed subtags) would mean that all vehicles (including buses, HGV, trailers etc.) can **also** go 60 mph, which is **incorrect**. So, one should _either_ add **all** implied maxspeed tags (if one thinks that having them explicitly tagged is beneficial for ease of use) _or_ **none** of them (if one thinks that implied defaults, and data-consumer-side processing of them is preferable).

But then there is also a question **is explicitly adding all those implied tags actually desirable?** It would certainly bloat the map, making processing slower and maps taking more space, introducing updating problem (as noted above) and QA issues (what to do when implicit and explicit data is out of sync), while not really bringing any new information on the table that isn't already implicitly known. Such _defaults_ better belong in data consumers instead of being explicitly tagged, IMHO. Much like adding `currency:gbp=yes` to every shop in GB would be useless, IMHO.
